### PR TITLE
Randr 1.5

### PR DESCRIPTION
**Pull request by jklong**
Addresses comments in #224 as it seems abandoned. Credit for most of the code here belongs to @allfro.

I expect 1.5 is more widespread now than when the comment was first made, but I added a quick version check regardless.

### CODE REVIEW

File: Xlib/ext/randr.py [hunk 1]
```diff
@@ -1186,12 +1267,20 @@ def init(disp, info):
...
+
+        disp.extension_add_event(info.first_event + RRScreenChangeNotify, ScreenChangeNotify)
+        # add RRNotify events (1 event code with 3 subcodes)
+        disp.extension_add_subevent(info.first_event + RRNotify, RRNotify_CrtcChange, CrtcChangeNotify)
+        disp.extension_add_subevent(info.first_event + RRNotify, RRNotify_OutputChange, OutputChangeNotify)
+        disp.extension_add_subevent(info.first_event + RRNotify, RRNotify_OutputProperty, OutputPropertyNotify)
+
+        #disp.extension_add_error(BadRROutput, BadRROutputError)
+        #disp.extension_add_error(BadRRCrtc, BadRRCrtcError)
+        #disp.extension_add_error(BadRRMode, BadRRModeError)
```
Comment (vasily-v-ryabov) [inline Xlib/ext/randr.py:1286]: What is the reason to keep this commented code? Please add a comment when it should be uncommented or please remove this code.

File: Xlib/protocol/rq.py [hunk 2]
```diff
@@ -604,7 +604,8 @@ def parse_value(self, val, display):
         return self.type.parse_value(val, display)
 
     def pack_value(self, val):
-        return self.type.pack_value(val)
+        val = self.type.pack_value(val)
+        return val, len(val), None
```
Comment (vasily-v-ryabov) [inline Xlib/protocol/rq.py:608]: This comment in PR #224 is not addressed. What is the reason for this change? I thought we should fix `pack_value` implementation for specific type `self.type`, not for general `Object`. Or please explain if I'm thinking wrong way.

**Review by vasily-v-ryabov — COMMENTED** [review]
Thanks for improving, it definitely looks better.

File: Xlib/protocol/rq.py (hunk 2 above)
Comment (vasily-v-ryabov) [reply Xlib/protocol/rq.py:608]: See #223 for more details of my suggestions.

Comment (vasily-v-ryabov) [reply Xlib/protocol/rq.py:608]: Maybe it's worth separating pull requests for Randr 1.5 and `pack_value`.

Comment (jklong) [reply Xlib/protocol/rq.py:608]: > Maybe it's worth separating pull requests for Randr 1.5 and `pack_value`.

Agreed. I'll pull this out.

File: Xlib/ext/randr.py (hunk 1 above)
Comment (jklong) [reply Xlib/ext/randr.py:1286]: I was hoping @allfro  would chime in for this and the `pack_value` changes. It seems the error classes referenced weren't defined, I've defined them and uncommented the code

**Review by vasily-v-ryabov — APPROVED** [review]

File: Xlib/protocol/rq.py (hunk 2 above)
Comment (sherbang) [reply Xlib/protocol/rq.py]: It looks like this change never was merged in a separate PR.

Without this change `set_monitor` is unusable, Ex:
```python
from Xlib import display
from Xlib.ext import randr

root = display.Display().screen().root
m = randr.get_monitors(root, is_active=True).monitors[0]
m.x = 500
randr.set_monitor(root, monitor_info=m)
```
raises "ValueError: too many values to unpack (expected 3)"

Also, more generally....
```python
from Xlib.protocol import rq
MonInfo = rq.Struct(
    rq.Int16('x'),
    rq.Int16('y'),
    rq.List('foo', rq.Int32),
)
TestStruct = rq.Struct(
    rq.Int16("a"),
    rq.Object('monitor_info', MonInfo),
)
TestStruct.to_binary(a=0, monitor_info=dict(x=1, y=0, foo=[1, 2, 3]))
```
fails in the same way.

The problem only occurs when the inner `Struct` contains a list.  If you remove the `List` from `MonInfo` then there's no exception.
